src/main/scala/coupledL2/GrantBuffer.scala

```diff
@@ -175,17 +218,18 @@ class GrantBuffer(implicit p: Parameters) extends L2Module {
 
   // =========== record unreceived GrantAck ===========
   // Addrs with Grant sent and GrantAck not received
-  val inflight_grant = RegInit(VecInit(Seq.fill(grantBufInflightSize){
-    0.U.asTypeOf(Valid(new InflightGrantEntry))
-  }))
   when (io.d_task.fire && dtaskOpcode(2, 1) === Grant(2, 1)) {
     // choose an empty entry
-    val insertIdx = PriorityEncoder(inflight_grant.map(!_.valid))
-    val entry = inflight_grant(insertIdx)
+    val entry = inflight_grant(inflight_insertIdx)
     entry.valid := true.B
     entry.bits.set    := io.d_task.bits.task.set
     entry.bits.tag    := io.d_task.bits.task.tag
-    entry.bits.sink   := io.d_task.bits.task.mshrId
+  } .elsewhen (io.d_task.fire && io.d_task.bits.task.mergeA) {
+    // if mergeA, alloc an entry for merge grant 
+    val entry = inflight_grant(inflight_insertIdx)
+    entry.valid := true.B
+    entry.bits.set := io.d_task.bits.task.set
+    entry.bits.tag := io.d_task.bits.task.tag
```

Comment:
is the logic is `when` and `elsewhen` **the same**?

Reply:
yes, I have merged them.